pkg/commands/git_commands/patch.go

```diff
@@ -132,17 +128,22 @@ func (self *PatchCommands) MovePatchToSelectedCommit(commits []*models.Commit, s
 		return err
 	}
 
+	patch, err := restorePatchFromOriginalCommit(self, commits[sourceCommitIdx])
+	if err != nil {
+		_ = self.rebase.AbortRebase()
+		return err
+	}
+
 	if self.rebase.onSuccessfulContinue != nil {
 		return errors.New("You are midway through another rebase operation. Please abort to start again")
 	}
 
 	self.rebase.onSuccessfulContinue = func() error {
 		// now we should be up to the destination, so let's apply forward these patches to that.
 		// ideally we would ensure we're on the right commit but I'm not sure if that check is necessary
-		if err := self.PatchManager.ApplyPatches(false); err != nil {
-			if err := self.rebase.AbortRebase(); err != nil {
-				return err
-			}
+		if err := self.rebase.workingTree.ApplyPatch(patch, "index", "3way"); err != nil {
+			// Don't abort the rebase here; this might cause conflicts, so give
+			// the user a chance to resolve them
```

Comment:
If the user does need to resolve conflicts here, then once they've resolved the conflicts the AmendHead part below won't happen because it's not wrapped in an 'onSuccessfulContinue' call. I don't think we need to solve that as part of this PR, but it's something we ought to fix at some point, perhaps with a better system for doing this stuff e.g. 'here's a sequence of actions we want to perform, and we might get conflicts any step of the way, so if that happens, allow the conflicts to be resolved, and then continue where we left off'.

Reply:
If they resolved and staged conflicts and simply continue, git will amend them into the last commit, which is what we want. This is a feature of git that I only learned about recently: if you "edit" a commit, then stage changes, then continue, git amends the changes into the last commit.

I think this is good enough, and we don't need to do anything on top of that.

Reply:
Oh wow, I didn't know it was that general. I thought the logic only applied to merge conflicts. Good to know.